clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesManifest.java

```diff
@@ -52,7 +52,7 @@ public KubernetesKind getKind() {
     // using ApiVersion here allows a translation from a kind of NetworkPolicy in the manifest to
     // something
     // like  NetworkPolicy.crd.projectcalico.org for custom resources
-    String kindName = getRequiredField(this, "kind");
+    String kindName = getKindName();
```

Comment:
Why did you pull this out into a function?

Reply:
I pulled this out to be public a method because I needed the kind name as a string instead of a `KubernetesKind` object, since there's no `KubernetesKind` for `List`. Also, using `getKind` on a non-registered kind has the effect of dynamically creating a new kind for it (used for custom resource definition objects), which then has other side effects like trying to cache those kinds.